src/lasp_dependence_dag.erl

```diff
@@ -44,20 +53,53 @@
 %% lets us quickly delete those edges.
 -type process_map() :: dict:dict(pid(), {id(), id()}).
 
+%% Stored metadata for a lasp process.
+%% We don't store the input(s) and output vertices as that information
+%% is implicitly stored by edges in the graph.
+-record(process_metadata, {read :: function(),
+                           transform :: function(),
+                           write :: function()}).
+
+%% We store a mapping Id -> {Pid, Path, MetadataList} to identify the list of
+%% vertices and relationships that get removed as part of a path contraction.
+%%
+%% Used during the cleaving step.
+-type contract_process_id() :: non_neg_integer().
```

Comment:
At first I was using the pid directly, but if the process is restarted by a supervisor, we have no way of knowing the new pid in order to update this information.

However, as far as I know, when a process is restarted, the supervisor uses exactly the same `init` arguments every time. This `contract_process_id` is just a hash of the read, transform and write functions so we can keep track of processes through restarts.

When a new edge is added to the graph, we check if we have previously seen the hash. If we have, then that process is a restart of an older one, and we know that we have to update the optimized map.

The `pid_table` is there to map pids to its hash. When a process is killed, we only know
its pid, so we have to get their hash in order to correctly remove the appropiate metadata from the `optimized_map`.